File: src/Scripting/Lua/patterns/distributed_proxy.lua
local remote = { fetch = function(id) return { id=id, value="ok" } end }
local proxy = { cache = {} }
function proxy:fetch(id)
  if not self.cache[id] then self.cache[id] = remote.fetch(id) end
  return self.cache[id]
end
local a, b = proxy:fetch("1"), proxy:fetch("1")
assert(a == b and a.value == "ok")
return true

File: src/Scripting/PHP/patterns/distributed_proxy.php
<?php declare(strict_types=1);
$check = static fn(bool $ok) => $ok ?: throw new RuntimeException('distributed_proxy');
$remote = static fn(string $json): string => json_encode(['result'=>json_decode($json,true)['x'] * 2]);
$proxy = static function(int $x) use ($remote): int { return json_decode($remote(json_encode(['x'=>$x])), true)['result']; };
$check($proxy(21) === 42);
